### PR TITLE
fix: added logic to hide the output and status of each cell/outline on edit while remaining the events list intact

### LINKED ISSUE #313: Workflow Cell output is cleared after editing code
This is likely being cleared when we calculate `isDirty` for the workflow, but we should leave it around as it is helpful in debugging.

My story:  I am trying to transform several columns in each row of the dataset, and am using `print()` to log my work.  I need to identify the columns so I can pull items out of a list, but when I start writing code, the output disappears and I need to dry run again to find the next column and continue writing code.

### PR DESCRIPTION
**Pull request by boandriy**
added logic to hide the output and status of each cell/outline on edit while remaining the events list intact
fixed the output clear action to hide the output of single cell

fixes #313

### CODE REVIEW

File: src/features/workflow/CodeEditor.tsx [hunk 1]
```diff
@@ -106,17 +108,20 @@ const CodeEditor: React.FC<CodeEditorProps> = ({
...
-    borderStyles = 'border-transparent hover:border-qritile border-2 border-solid'
+  let borderStyles = 'border-transparent hover:border-qritile border-2 border-solid';
+  if(!isEdited){
+    if(status === 'succeeded') {
+      borderStyles = `border-qrigreen border-2 border-solid ${hasOutput && 'border-b-0'}`
+    }else if(status === 'running'){
+      borderStyles = 'border-qrinavy-700 border-2 border-solid border-b-0 -mb-1'
+    }else if(status === 'failed'){
+      borderStyles = `border-dangerred border-2 border-solid ${hasOutput && 'border-b-0'}`
+    }else{
```
Comment (ramfox) [inline src/features/workflow/CodeEditor.tsx]: Based on chris' answer on the issue I think the logic here isn't right (here is the comment: https://github.com/qri-io/frontend/issues/313#issuecomment-929369275)

When any cell as been edited, the *sidebar* and *all of the cells* should not have a "success" border any more & show a symbol that indicates that it is not in a valid state. In this comment, he mentions using the gray circle with the line as the symbol: https://github.com/qri-io/frontend/issues/313#issuecomment-929380395

**Review by ramfox — CHANGES_REQUESTED** [review]
everything looks good, but I think the logic for the cell outlines is off, based on the comment's chris left on the issue

**Comment by chriswhong** [comment]
image

When testing this locally, I see this green outline appearing on hover.  If this is the "active" state of the cell, it should not be related to hover and should just appear this way when the user has clicked it it and is ready to edit, regardless of whether they have made changes.

 On this note, we need a way for the current cell to stay active AND to show that its last run was successful.

On deepnote, they use the icon to indicate success, but there is no outline color for success/failure.  The outline is only used for the "active" cell that the user is interacting with.

When the user edits the cell, the results stay, but the icon disappears.

**Comment by b5** [comment]
where are we at with this? need re-review? @chriswhong have you confirmed it works as expected?

**Comment by chriswhong** [comment]
Yes, I pushed changes that are working as expected, just need the ok on code quality/maintainability

File: src/features/workflow/state/workflowState.ts [hunk 2]
```diff
@@ -76,6 +85,12 @@ export interface WorkflowState {
   // stores the "clean" state of triggers, steps, and hooks, used to compare
   // with working state to determine isDirty
   workflowBase: WorkflowBase
+  // stores a boolean value for each workflow code cell to represent if
+  // cell was edited
+  areCellsEdited: boolean[]
```
Comment (b5) [inline src/features/workflow/state/workflowState.ts]: don't use `are` in a field name: `areCellsEdited` -> `editedCells`

File: src/features/workflow/state/workflowState.ts [hunk 3]
```diff
@@ -76,6 +85,12 @@ export interface WorkflowState {
...
+  // stores a boolean value for each workflow code cell to represent if
+  // cell output was cleared
+  areCellsOutputCleared: boolean[]
```
Comment (b5) [inline src/features/workflow/state/workflowState.ts]: `areCellsOutpuCleared` -> `clearedOutputCells`

File: src/features/workflow/WorkflowCell.tsx [hunk 4]
```diff
@@ -1,13 +1,17 @@
 import React from 'react'
+import classNames from 'classnames'
 import { RunStep } from '../../qri/run'
 import { TransformStep } from '../../qri/dataset'
 import CodeEditor from './CodeEditor'
 import Output from './output/Output'
 import ScrollAnchor from '../scroller/ScrollAnchor'
 import WorkflowHeader from './WorkflowHeader'
 import WorkflowCellControls from './WorkflowCellControls'
+import { useSelector } from "react-redux";
```
Comment (b5) [inline src/features/workflow/WorkflowCell.tsx]: keep npm / external package imports grouped at the top

**Pull request merged**